vars/writeBuildInfo.groovy

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/groovy
+
+import groovy.json.JsonBuilder
+import groovy.json.JsonSlurperClassic
+import org.feedhenry.Utils
+
+def call(name, versionTxt) {
+    def utils = new Utils()
+    def buildInfoFileName = utils.getBuildInfoFileName()
+    def buildInfo = [:]
+
+    if(fileExists(buildInfoFileName)) {
+        def buildInfoRaw = readFile buildInfoFileName
+        print buildInfoRaw
+        buildInfo = new JsonSlurperClassic().parseText buildInfoRaw
+        print buildInfo
+    }
+
+    buildInfo['jenkinsUrl'] = env.JENKINS_URL
+    buildInfo['buildUrl'] = env.BUILD_URL
+    buildInfo['sha1'] = sh(returnStdout: true, script: "git log -n 1 --pretty=format:\"%H\"").trim()
+
+    buildInfo[name] = [:]
+    buildInfo[name]['version'] = versionTxt.split('-')[0]
+    buildInfo[name]['build'] = env.BUILD_NUMBER
```

Comment:
If we can always say that the 'build' property will be the bit after the  '-' in versionTxt (I'm not sure we can, esp. when we get to module/library projects), then you could do something like this here:

```groovy
buildInfo[name] = [:]
buildInfo[name].with {
  (version, build) = versionTxt.split('-')
}
```

I'm not sure if that would even be better or not. See here for more details on the multiple assignment, if interested: https://objectpartners.com/2014/07/09/groovys-with-and-multiple-assignment/